app/controllers/api/test_script_results_controller.rb

```diff
@@ -0,0 +1,190 @@
+                                          'submission',
+                                          assignment,
+                                          grouping,
+                                          submission)
+        render 'shared/http_status', locals: {code: '201', message:
+          HttpStatusHelper::ERROR_CODE['message']['201']}, status: 201
+      else
+        # Some other error occurred
+        render 'shared/http_status', locals: { code: '500', message:
+          HttpStatusHelper::ERROR_CODE['message']['500'] }, status: 500
+      end
+    end
+
+    # Deletes a Test Script Result instance
+    # Requires: assignment_id, group_id, id
+    def destroy
+      # get_submission renders appropriate error if the submission isn't found
+      submission = get_submission(params[:assignment_id], params[:group_id])
+      return if submission.nil?
+
+      test_script_result = submission
+                           .test_script_results
+                           .find_by_id(params[:id])
+
+      # Render error if the TestResult does not exist
+      if test_script_result.nil?
+        render 'shared/http_status', locals: { code: '404', message:
+          'Test script result was not found'}, status: 404
+        return
+      end
+
+      if test_script_result.destroy
+        # Successfully deleted the TestResult; render success
+        render 'shared/http_status', locals: { code: '200', message:
+          HttpStatusHelper::ERROR_CODE['message']['200']}, status: 200
+      else
+        # Some other error occurred
+        render 'shared/http_status', locals: { code: '500', message:
+          HttpStatusHelper::ERROR_CODE['message']['500'] }, status: 500
+      end
+    end
+
+    # Updates a Test Script Result instance. Deletes the current test script
+    #  result and its test results and reprocess the xml test harness file. 
+    # This is basically a delete followed by a create
+    # Requires: assignment_id, group_id, id
+    # Optional:
+    #  - file_content: New contents of the test results file
+    def update
+      if has_missing_params?([:file_content])
+        # incomplete/invalid HTTP params
+        render 'shared/http_status', locals: {code: '422', message:
+          HttpStatusHelper::ERROR_CODE['message']['422']}, status: 422
+        return
+      end
+
+      # get_submission renders appropriate error if the submission isn't found
+      submission = get_submission(params[:assignment_id], params[:group_id])
+      return if submission.nil?
+
+      test_script_result = submission.test_script_results.find_by_id(params[:id])
+
+      # Render error if the Test Script Result does not exist
+      if test_script_result.nil?
+        render 'shared/http_status', locals: { code: '404', message:
+          'Test script result was not found'}, status: 404
+        return
+      end
+
+      grouping = submission.grouping
+      assignment = submission.assignment
+
+      if test_script_result.destroy &&
```

Comment:
I'm not sure this is the best approach. If there's an error processing the new params, won't the existing test script result be removed?

I suppose you could process the params first, and then if that succeeds to the replacement.